### PR TITLE
TYP: activate Check for missing error codes

### PR DESCRIPTION
**Pull request by simonjayhawkins**
xref #35311

### CODE REVIEW

**Review by WillAyd — APPROVED** [review]
lgtm - nice work!

File: ci/code_checks.sh [hunk 1]
```diff
@@ -230,10 +230,9 @@ if [[ -z "$CHECK" || "$CHECK" == "patterns" ]]; then
     invgrep -R --include="*.py" -P '# type: (?!ignore)' pandas
     RET=$(($RET + $?)) ; echo $MSG "DONE"
 
-    # https://github.com/python/mypy/issues/7384
-    # MSG='Check for missing error codes with # type: ignore' ; echo $MSG
-    # invgrep -R --include="*.py" -P '# type: ignore(?!\[)' pandas
-    # RET=$(($RET + $?)) ; echo $MSG "DONE"
+    MSG='Check for missing error codes with # type: ignore' ; echo $MSG
+    invgrep -R --include="*.py" -P '# type: ignore(?!\[)' pandas
```
Comment (jbrockmendel) [inline ci/code_checks.sh:234]: should we check for `# type:ignore` without the space?  we have one of those in plotting._matplotlib.core

Comment (simonjayhawkins) [reply ci/code_checks.sh]: i've updated the check to account for optional space. We could also consider preventing that pattern.

**Comment by jreback** [comment]
thanks @simonjayhawkins

**Pull request merged**